clang/test/CIR/Lowering/vectype-cl.cpp

```diff
@@ -0,0 +1,53 @@
+// RUN: %clang_cc1 -std=c++17 -triple x86_64-unknown-linux-gnu -fclangir -emit-cir %s -o %t.cir
```

Comment:
Nothing opencl specific that we should use here? If we are testing stuff against regular linux do we need to mention opencl at all?

Reply:
Sorry for the confusing description on the top! In clang, the OpenCL vectors are "implemented" by extended vectors (`__attribute__((ext_vector_type(4)))`), which supports extra syntax like swizzles. Extended vectors are a "one-for-all" implementation of OpenCL vectors, Apple Metal vectors, etc. We don't support OpenCL langOpts or SPIR-V target currently, therefore x86_64 is intentionally used as a secondary option. If we don't touch the backend, then all code paths are the same. You remind me that this file is actually a test for `ext_vector_type`😂.

If it looks too strange, we may defer this PR until the initial support for OpenCL lands.

Reply:
Right, exactly because it's not opencl only that I'm asking that the support is added in terms of `ext_vector_type` instead! It doesn't look strange, we just need some renaming :)